packages/core/workflows-sdk/src/utils/composer/type.ts

```diff
@@ -62,7 +64,9 @@ export type StepFunction<
 > = (KeysOfUnion<TInput> extends []
   ? // Function that doesn't expect any input
     {
-      (): WorkflowData<TOutput> & StepFunctionReturnConfig<TOutput>
+      (): TOutput & {} extends never
```

Comment:
Is the `& {}` necessary? I am just not sure what purpose it has :D

Reply:
The goal here is to check if merging `TOutput` (aka step output) with an object produces `never`, then we should treat that output differently. 

It will produce `never` when the return value is `undefined` or `null`. But not a union of `undefined + something else`.

Reply:
👍  TIL